### PR TITLE
Add delete option to proposition document

### PR DESCRIPTION
**Pull request by bau123**
Add delete for proposition document. This is to help users with document upload fails to delete and retry.

## Description of change

Add delete option for proposition documents

## Test instructions
Go to an investment
Select any proposition
Upload a document
Delete Document
Redirect back to proposition 

### Before
image

### After
image

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [ ] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

**Review by dredmonds — APPROVED** [review]

File: src/client/modules/Investments/Projects/Propositions/DeletePropositionDocument.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+const DeletePropositionDocument = () => {
+  const { projectId, propositionId, documentId } = useParams()
+  return (
+    <InvestmentResource id={projectId}>
+      {(project) => (
+        <DefaultLayout
+          heading={'Deleting proposition document'}
+          pageTitle={`Delete proposition document - ${project.name} - Projects - Investments`}
+          breadcrumbs={buildProjectBreadcrumbs([
```
Comment (DeanElliott96) [inline src/client/modules/Investments/Projects/Propositions/DeletePropositionDocument.jsx:22]: You could add some tests here for the breadcrumbs, I've seen a lot of these kind of tests.

File: src/client/modules/Investments/Projects/Propositions/transformers.js [hunk 2]
```diff
@@ -45,25 +48,43 @@ export const transformAbandonedPropositionForAPI = ({
...
+      >
+        Download
+      </Link>
+    ) : (
+      <strong>
+        The file didn't pass virus scanning, contact your administrator
+      </strong>
+    )}
+    <StyledSpan>
+      <Link
```
Comment (DeanElliott96) [inline src/client/modules/Investments/Projects/Propositions/transformers.js:75]: Maybe another test to assert the link is what you expect it to be?

**Review by DeanElliott96 — COMMENTED** [review]
Looks good, could maybe add some tests?

**Review by DeanElliott96 — APPROVED** [review]
Accepting and to look at tests together tomorrow.

**Pull request merged**